recipes-app/iot2050-efivarfs-helper/files/iot2050-efivarfs-helper.tmpl

```diff
@@ -0,0 +1,180 @@
+import sys
+import argparse
+import subprocess
+
+
+__version__ = "${PV}"
+
+
+EFIVARFS_MOUNT_POINT = '/sys/firmware/efi/efivars'
+
+
+def err_print(*args, **kwargs):
+    print(*args, file=sys.stderr, **kwargs)
+
+
+def exit_on_failure(message, code=1):
+    err_print()
+    err_print(message)
+    err_print()
+    exit(code)
+
+
+def dump_subprocess_error(e: subprocess.CalledProcessError):
+    err_print()
+    err_print("ERROR:\t" + str(e).strip())
+    stdout = str(e.stdout).strip()
+    stderr = str(e.stderr).strip()
+    if stdout != '':
+        err_print("ERROR:\tSTDOUT: " + stdout)
+    if stderr != '':
+        err_print("ERROR:\tSTDERR: " + stderr)
+
+
+def change_efivarfs_immutable_flag(immutable: bool):
+    if immutable:
+        action = 'Adding'
+        option = '+i'
+    else:
+        action = 'Removing'
+        option = '-i'
+
+    for file in os.listdir(EFIVARFS_MOUNT_POINT):
+        if (file.startswith('db-')
+                or file.startswith('KEK-')
+                or file.startswith('PK-')):
+            filepath = os.path.join(EFIVARFS_MOUNT_POINT, file)
+            print("{} the immutable flag of {}... ".format(action, filepath),
+                  end='')
+            cmd = 'chattr {} {}'.format(option, filepath)
+            result = subprocess.run(
+                cmd, shell=True, capture_output=True, check=True, text=True)
+            print("Done.")
+
+
+def do_prepare():
+    message_prepare_fail = "Fail to prepare the efivarfs for manipulating. \
+You may need to check the environment."
+
+    try:
+        mod_required = 'tee_stmm_efi'
+        print("Checking for the kernel module {}... ".format(mod_required),
+              end='')
+        cmd = 'lsmod'
+        result = subprocess.run(
+            cmd, shell=True, capture_output=True, check=True, text=True)
+        if len([i for i in result.stdout.split('\n')
+                if i.startswith("{} ".format(mod_required))]) == 1:
+            print("Done.")
+        else:
+            print("ERROR: {} is not loaded!".format(mod_required))
+            exit_on_failure(message_prepare_fail)
+    except subprocess.CalledProcessError as e:
+        dump_subprocess_error(e)
+        exit_on_failure(message_prepare_fail, code=e.returncode)
+
+    # tee-supplicant is used to serve the trusted applications running in
+    # optee-os for accessing RPMB partition.
+    try:
+        print("Checking for the service tee-supplicant... ", end='')
+        cmd = 'systemctl is-active tee-supplicant.service'
+        result = subprocess.run(
+            cmd, shell=True, capture_output=True, check=True, text=True)
+
+        print("Done.")
+    except subprocess.CalledProcessError as e:
+        print("ERROR: service is inactive!")
+        exit_on_failure(message_prepare_fail)
+
+    # efivarfs must be mounted with read & write option.
+    try:
+        print("Checking for efivarfs mounting... ", end='')
+        cmd = 'findmnt {}'.format(EFIVARFS_MOUNT_POINT)
```

Comment:
Maybe better done with the included batteries: [os.path.ismount](https://docs.python.org/3/library/os.path.html#os.path.ismount)